### PR TITLE
Add TestU01_stdin and generate

### PR DESCRIPTION
**Pull request by curiousleo**
A few open question inline.

### CODE REVIEW

File: testu01-stdin/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    gcc -std=c99 -O3 -fomit-frame-pointer -mtune=native TestU01_stdin.c -o TestU01_stdin -ltestu01 -lprobdist -lmylib -lm
+  '';
+  installPhase = ''
+    mkdir -p $out/bin
+    cp TestU01_stdin $out/bin
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Empirical Statistical Testing of Uniform Random Number Generators";
+    license = licenses.gpl3;
```
Comment (curiousleo) [inline testu01-stdin/default.nix:21]: @idontgetoutmuch, do you know what the license (and homepage) of `TestU01_stdin.c` are?

File: shell.nix [hunk 2]
```diff
@@ -1,22 +1,26 @@
-let overlay = self: super:
```
Comment (curiousleo) [inline shell.nix:1]: Formatted with `nixpkgs-fmt`, added `generate` and `testu01-stdin`, removed `libiconv` (don't know what that is).

File: testu01-stdin/TestU01_stdin.c [hunk 3]
```diff
@@ -0,0 +1,211 @@
+/*
```
Comment (curiousleo) [inline testu01-stdin/TestU01_stdin.c:1]: I wasn't really sure where to put this ... opted for a new folder `testu01-stdin`, but no strong opinion on location / naming.

File: testu01-stdin/default.nix (hunk 1 above)
Comment (idontgetoutmuch) [reply testu01-stdin/default.nix:21]: They have disappeared. I found the code cached by google: http://webcache.googleusercontent.com/search?q=cache:Sw4Wm-UXUjsJ:www.freecx.co.uk/crypto/cryptanalysis/TestTools/TestU01/TestU01_stdin.c+&cd=2&hl=en&ct=clnk&gl=uk.

I guess we could write to the author?

File: shell.nix (hunk 2 above)
Comment (idontgetoutmuch) [reply shell.nix:1]: Nooo - I need libiconv on macos

File: testu01-stdin/TestU01_stdin.c (hunk 3 above)
Comment (idontgetoutmuch) [reply testu01-stdin/TestU01_stdin.c:1]: Fine with me

File: shell.nix (hunk 2 above)
Comment (curiousleo) [reply shell.nix:1]: I'll add it back then!

File: testu01-stdin/default.nix (hunk 1 above)
Comment (idontgetoutmuch) [reply testu01-stdin/default.nix:21]: No luck - they have covered their tracks well

File: shell.nix (hunk 2 above)
Comment (curiousleo) [reply shell.nix:1]: Done

**Pull request merged**